### PR TITLE
readme updated with qutip.devmajor and cupy instructions

### PR DESCRIPTION
**Pull request by MrRobot2211**
This updates the readme with an step by step explanation of how to install the unreleased version and required libraries,  Just for adventurers.

See https://github.com/qutip/qutip-cupy/issues/14. About usage, maybe we should point to the data_layer documentation. In any case there is not much usage we can reap from it now, but I have a basic notebook with examples already is there a correct place to hang it?

### CODE REVIEW

**Comment by nwlambert** [comment]
For what its worth, this worked without much fuss in windows 👍

File: README.md [hunk 1]
```diff
@@ -19,3 +19,18 @@ Installation
...
+- conda install cudatoolkit
+- conda install -c conda-forge cupy
+
+To install `QuTiP >5.0.0` while it is not yet released we recommend
+
+- python -mpip install git+https://github.com/qutip/qutip.git@dev.major
+
+Now you can safely install `qutip-cupy`
+
+- python -mpip install https://github.com/qutip/qutip-cupy.git
```
Comment (nwlambert) [inline README.md:36]: maybe this needs a git+ prefix?

Comment (MrRobot2211) [reply README.md:36]: Good catch

File: README.md [hunk 2]
```diff
@@ -19,3 +19,18 @@ Installation
 ------------
 
 `qutip-cupy` is not yet officially released.
+
+If you want to try out the package you will need to have a CUDA enabled GPU, `QuTiP >5.0.0` and `CuPy`.
+We recommend using a conda environment `Python >= 3.7`.
+To install `CuPy` we recommend the following steps:
+
+- conda install cudatoolkit
```
Comment (leofang) [inline README.md:27]: This line is redundant as `cudatoolkit` is a dependency of `cupy` which gets handled in the next line.
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -19,3 +19,18 @@ Installation
...
+- conda install -c conda-forge cupy
+
+To install `QuTiP >5.0.0` while it is not yet released we recommend:
+
+- python -mpip install git+https://github.com/qutip/qutip.git@dev.major
```
Comment (leofang) [inline README.md:32]: Just a nitpick: I don't like the Linux style (or whatever you call) argument invocation 😅
```suggestion
- python -m pip install git+https://github.com/qutip/qutip.git@dev.major
```

File: README.md [hunk 4]
```diff
@@ -19,3 +19,18 @@ Installation
...
+
+Now you can safely install `qutip_cupy`
+
+- python -mpip install git+https://github.com/qutip/qutip-cupy.git
```
Comment (leofang) [inline README.md:36]: Ditto
```suggestion
- python -m pip install git+https://github.com/qutip/qutip-cupy.git
```

File: README.md (hunk 3 above)
Comment (MrRobot2211) [reply README.md:32]: We have that type of invocations all over the `yml`s

Comment (leofang) [reply README.md:31]: Ah, ok, let's ignore this and make the style consistent then.

Comment (hodgestar) [reply README.md:31]: I find the `-mpip` weird too, but as @MrRobot2211 says, it does appear to be fairy ubiquitous these days. ;)

**Review by hodgestar — APPROVED** [review]

**Pull request merged**